stringfuzz/generators/regex.py

```diff
@@ -70,27 +100,42 @@ def make_regex_string(min_length, max_length):
 
     return smt_str_to_re(smt_str_lit(string))
 
-def make_random_term(depth):
+def make_random_term(depth, operator_index):
     if depth == 0:
         return make_regex_string(_literal_min, _literal_max)
 
-    subterm = make_random_term(depth - 1)
+    if _operator_type == OPERATOR_ALTERNATING:
+        next_operator_index = operator_index + 1
+    else:
+        next_operator_index = random.randrange(len(_operator_list))
 
-    # randomly pick a recursive regex term
-    random_result = random.randint(0, 2)
+    operator = get_operator_at_index(operator_index)
+    subterm = make_random_term(depth - 1, next_operator_index)
 
-    if random_result == 0:
+    if operator == OPERATOR_STAR:
         return smt_regex_star(subterm)
 
-    elif random_result == 1:
+    if operator == OPERATOR_PLUS:
         return smt_regex_plus(subterm)
 
-    elif random_result == 2:
-        second_subterm = make_random_term(depth - 1)
+    if operator == OPERATOR_UNION:
+        second_subterm = make_random_term(depth - 1, next_operator_index)
         return smt_regex_union(subterm, second_subterm)
 
+    if operator == OPERATOR_INTER:
+        second_subterm = make_random_term(depth - 1, next_operator_index)
+        return smt_regex_inter(subterm, second_subterm)
+
+    if operator == OPERATOR_CONCAT:
+        second_subterm = make_random_term(depth - 1, next_operator_index)
+        return smt_regex_concat(subterm, second_subterm)
+
 def make_random_terms(num_terms, depth):
-    terms = [make_random_term(depth) for i in range(num_terms)]
+    if _operator_type == OPERATOR_ALTERNATING:
+        terms = [make_random_term(depth, 0) for i in range(num_terms)]
+    else:
+        terms = [make_random_term(depth, random.randrange(len(_operator_list))) for i in range(num_terms)]
```

Comment:
Small nitpick: it would be more readable to create a `starting_operator` variable that's `0` for the first case and `random.randrange(len(_operator_list))` for the second case, and then have this only once:
```
terms = [make_random_term(depth, starting_operator) for i in range(num_terms)]
```

Reply:
But wouldn't this result in each term having the same starting operator?